### PR TITLE
[NativeAOT] Creation of non-suspendable threads does not need to synchronize with the caller.

### PR DESCRIPTION
**Pull request by VSadov**
Unlike suspendable threads that attach themselves to the threadstore without taking locks (and thus must be initialized sequentially one-by-one), unsuspendable threads have very trivial self-contained initialization and can be created in a shoot-and-forget way. 

On server GC we create #procs of GC threads early in startup and waiting for each of them to become runnable and able to communicate that to the caller before we start creating another is an unnecessary delay. 

On my machine (Windows10, AMD 5950X, 16 cores, 32 logical) this adds ~20% (1.1 msec.) to the time spent in `InitializeRuntime`.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/gcenv.ee.cpp [hunk 1]
```diff
@@ -533,54 +533,84 @@ struct ThreadStubArguments
...
-    ThreadStubArguments threadStubArgs;
+    if (!is_suspendable)
+        return CreateUnsuspendableThread(threadStart, arg, name);
 
+    ThreadStubArguments threadStubArgs;
     threadStubArgs.m_pRealStartRoutine = threadStart;
     threadStubArgs.m_pRealContext = arg;
-    threadStubArgs.m_isSuspendable = is_suspendable;
 
     if (!threadStubArgs.m_ThreadStartedEvent.CreateAutoEventNoThrow(false))
```
Comment (VSadov) [inline src/coreclr/nativeaot/Runtime/gcenv.ee.cpp:582]: We'd also avoid creating and destroying #proc of these events at startup.

File: src/coreclr/nativeaot/Runtime/gcenv.ee.cpp [hunk 2]
```diff
@@ -533,54 +533,84 @@ struct ThreadStubArguments
 {
     void (*m_pRealStartRoutine)(void*);
     void* m_pRealContext;
-    bool m_isSuspendable;
     CLREventStatic m_ThreadStartedEvent;
 };
 
+static bool CreateUnsuspendableThread(void (*threadStart)(void*), void* arg, const char* name)
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/gcenv.ee.cpp:539]:
```suggestion
static bool CreateNonSuspendableThread(void (*threadStart)(void*), void* arg, const char* name)
```

Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/gcenv.ee.cpp:539]: I think it is better English, and it matches the equivalent method name in regular CoreCLR.

**Review by jkotas — APPROVED** [review]
LGTM otherwise. Thank you!

**Comment by VSadov** [comment]
Thanks!

**Pull request merged**